api/operations/approve_role_request.py

```diff
@@ -0,0 +1,151 @@
+from api.plugins import get_notification_hook
+from api.views.schemas import AuditLogSchema, EventType
+
+
+class ApproveRoleRequest:
+    def __init__(
+        self,
+        *,
+        role_request: RoleRequest | str,
+        approver_user: Optional[OktaUser | str] = None,
+        approval_reason: str = "",
+        ending_at: Optional[datetime] = None,
+        notify: bool = True,
+    ):
+        self.role_request = (
+            RoleRequest.query.options(
+                joinedload(RoleRequest.active_requested_group), joinedload(RoleRequest.active_requester_role)
+            )
+            .filter(RoleRequest.id == (role_request if isinstance(role_request, str) else role_request.id))
+            .first()
+        )
+
+        if approver_user is None:
+            self.approver_id = None
+            self.approver_email = None
+        elif isinstance(approver_user, str):
+            approver = db.session.get(OktaUser, approver_user)
+            self.approver_id = approver.id
+            self.approver_email = approver.email
+        else:
+            self.approver_id = approver_user.id
+            self.approver_email = approver_user.email
+
+        self.approval_reason = approval_reason
+
+        self.ending_at = ending_at
+
+        self.notify = notify
+
+        self.notification_hook = get_notification_hook()
+
+    def execute(self) -> RoleRequest:
+        # Don't allow approving a request that is already resolved
+        if self.role_request.status != AccessRequestStatus.PENDING or self.role_request.resolved_at is not None:
+            return self.role_request
+
+        # Don't allow requester to approve their own request
+        if self.role_request.requester_user_id == self.approver_id:
+            return self.role_request
+
+        # Don't allow approving a request if the reason is invalid and required
+        valid, _ = CheckForReason(
+            group=self.role_request.requester_role_id,
+            reason=self.approval_reason,
+            members_to_add=[self.role_request.requested_group_id] if not self.role_request.request_ownership else [],
+            owners_to_add=[self.role_request.requested_group_id] if self.role_request.request_ownership else [],
+        ).execute_for_role()
+        if not valid:
+            return self.role_request
+
+        # Don't allow approving a request if the requester role is deleted
+        requester = db.session.get(RoleGroup, self.role_request.requester_role_id)
+        if requester is None or requester.deleted_at is not None:
+            return self.role_request
+
+        # Don't allow approving a request for an a deleted or unmanaged group
+        if self.role_request.active_requested_group is None:
+            return self.role_request
+        if not self.role_request.active_requested_group.is_managed:
+            return self.role_request
+
+        self.role_request.status = AccessRequestStatus.APPROVED
```

Comment:
Should this be done inside ModifyRoleGroups? That way whenever ModifyRoleGroups is called (eg. an admin adds the role to the group manually), the corresponding request is also automatically approved.